design/general-global-rate-limit-policy.md

```diff
@@ -0,0 +1,100 @@
+# General Global RateLimit Policy
+
+Status: Draft
+
+## Abstract
+Define a general rate limit policy in the Contour configuration to be used as a global rate limit policy by all HTTPProxy.
+
+## Background
+See https://github.com/projectcontour/contour/issues/5357 for context.
+
+## Goals
+- Define a general L7 global rate limit policy which is used by all HTTPProxy on the `VirtualHost` level.
+
+## Non Goals
+- General rate limit policy for Route level `rateLimitPolicy`.
+- L4 rate limiting support. 
+
+## High-Level Design
+Currently, the global `rateLimitPolicy` for `VirtualHost` is set per HTTPProxy. This option allows each service to configure its own global `rateLimitPolicy`. However, for a single tenant setup/general usecase, this means you have to edit all the HTTPProxy objects with the same ratelimit policies.
+
+A new field, `GeneralRateLimitPolicy` (optional), will be added to `RateLimitServiceConfig` which is part of `ContourConfigurationSpec`. This field will define rate limit descriptors that will be added to every `VirtualHost`.
```

Comment:
I'm also thinking about whether it'd make sense to put this new field under the `Policy` field in config which we document as "default policy applied if not overridden by the user", since it's not strictly a property of the rate limit service itself, and does logically make sense under `Policy`. The downside is that rate limit-related config is split across two separate places. Interested in others' thoughts here.

Reply:
Ahh, I didn't actually notice the `Policy` section, thanks for pointing it out. 
So from a developer's perspective, I would like to configure everything related to rate limiting in one place. It is a bit more intuitive to do it in the `rateLimitServiceConfig` that `Policy`
But no strong opinion on this.

Reply:
Let's see if @sunjayBhatia @tsaarni have any thoughts here before we make a final decision.

Reply:
seems like we're a little inconsistent here, for global extAuth we have `AuthorizationServer` which contains the `AuthPolicy` field and some others not specifically about the auth server

Reply:
The generic `Policy` type has also field [`applyToIngress`](https://github.com/projectcontour/contour/blob/b5469efa7d6725a9e182fd226e333adc1337174e/apis/projectcontour/v1alpha1/contourconfig.go#L725-L729), which currently means that (header) policies are applied to `Ingresses`. If adding rate limiting to the type, should that too apply to `Ingresses`?